content/livesync/models/models.textile

```diff
@@ -0,0 +1,278 @@
+---
+title: Models
+meta_description: "A model is a live, observable data model backed by your database that you can use to keep your frontend applications updated in realtime with the latest information from your database."
+product: livesync
+languages:
+  - javascript
+---
+
+A model is a single instance of a live, observable data model backed by your database. A model requires a "sync function":#sync to tell the model how to initialize with the latest data, and a "merge function":#merge to calculate the next version of the model state when change events are received from the backend database.
+
+You can then "subscribe":#subscribe to a model to receive updates in realtime whenever the model state changes.
+
+h2(#create). Create or retrieve a model
+
+To create the model, use the @models.get()@ method on the client. If a model with the given name already exists, it will be returned.
+
+To instantiate a model you must provide a unique name. This identifies the model on the client, and is also the name of the "channel":/channels used to subscribe to state updates from the backend.
+
+```[javascript]
+const modelsClient = modelsClient.models.get({
+  name: 'myPost',
+  channelName: 'post:123',
+  sync,
+  merge,
+});
+```
+
+<aside data-type='note'>
+<p>Restrictions for model names:</p>
```

Comment:
We don't really have any restrictions on the model name, it's just a way to identify it by some string on the client. The channel name has the usual restrictions though.

Reply:
Good to know, I've removed that in 5df7fc28f18f1034c61584cd428bdf64c99d867e